workshop/permissions.py

```diff
@@ -61,3 +57,19 @@ def has_permission(self, request, view):
         if profile.get_club_privileges() or profile.get_workshop_privileges():
             return True
         return False
+
+class AllowWorkshopHeadOrContactForResource(permissions.BasePermission):
+    message = "You are not authorized to perform this task"
+
+    def has_object_permission(self, request, view, obj):
+        if request.user.is_anonymous:
+            return False
```

Comment:
```suggestion
        if request.method in permissions.SAFE_METHODS:
            return True
        if request.user.is_anonymous:
            return False
```

Reply:
I think this must be added, otherwise in `/resources/{id}/` `GET` request, a user will receive `401 Unauthorized Error` if he doesn't fulfill the permissions criteria, i.e. if he isn't a Workshop Head or Contact. Please test this endpoint too!

Reply:
@nishantwrp @krashish8 Prefer using [`is_authenticated`](https://docs.djangoproject.com/en/3.0/ref/contrib/auth/#django.contrib.auth.models.User.is_authenticated).
Check out the why here - https://code.djangoproject.com/ticket/2332

Reply:
Okay. Got it!